### PR TITLE
Add wait for max stake command

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Bash logic of wait for stake to come down below 66% is broken and causing all the automation tests to fail.

#### Summary of Changes

Implement wait-for-max-stake as a CLI function and use that for the automation script.

Fixes #

### CODE REVIEW

File: system-test/automation_utils.sh [hunk 1]
```diff
@@ -69,17 +69,9 @@ function wait_for_bootstrap_validator_stake_drop {
...
   # shellcheck disable=SC2154
-    bootstrap_validator_validator_info="$(ssh "${sshOptions[@]}" "${validatorIpList[0]}" '$HOME/.cargo/bin/solana --url http://127.0.0.1:8899 validators | grep "$($HOME/.cargo/bin/solana-keygen pubkey ~/solana/config/bootstrap-validator/identity.json)"')"
-    bootstrap_validator_stake_percentage="$(echo "$bootstrap_validator_validator_info" | awk '{gsub(/[\(,\),\%]/,""); print $9}')"
-
-    if [[ $(echo "$bootstrap_validator_stake_percentage < $max_stake" | bc) -ne 0 ]]; then
-      echo "Bootstrap validator stake has fallen below $max_stake to $bootstrap_validator_stake_percentage"
-      break
-    fi
-    echo "Max bootstrap validator stake: $max_stake.  Current stake: $bootstrap_validator_stake_percentage.  Sleeping 30s for stake to distribute."
-    sleep 30
```
Comment (mvines) [inline system-test/automation_utils.sh:82]: 👢 glad to see this go

Comment (t-nelson) [reply system-test/automation_utils.sh:82]: Same :tada: 

Though I thought I replaced that awk/gsub with something more resilient a few weeks ago :thinking:

File: system-test/automation_utils.sh [hunk 2]
```diff
@@ -69,18 +69,7 @@ function wait_for_bootstrap_validator_stake_drop {
...
-  done
+  ssh "${sshOptions[@]}" "${validatorIpList[0]}" '$HOME/.cargo/bin/solana wait-for-max-stake 66 --url http://127.0.0.1:8899'
```
Comment (t-nelson) [inline system-test/automation_utils.sh]: s/66/$max_stake/ ?

File: cli/src/cli.rs [hunk 3]
```diff
@@ -99,6 +99,9 @@ pub enum CliCommand {
         seed: String,
         program_id: Pubkey,
     },
+    WaitForMaxStake {
+        max_stake_percent: f32,
+    },
```
Comment (CriesofCarrots) [inline cli/src/cli.rs:104]: Can you please A-Z these additions within the cluster-query sections?

File: client/src/rpc_client.rs [hunk 4]
```diff
@@ -353,6 +353,38 @@ impl RpcClient {
         self.send(RpcRequest::GetVoteAccounts, json!([commitment_config]))
     }
 
+    pub fn wait_for_max_stake(
+        &self,
+        commitment: CommitmentConfig,
+        max_stake_percent: f32,
+    ) -> ClientResult<()> {
+        let mut current_percent;
+        loop {
```
Comment (t-nelson) [inline client/src/rpc_client.rs:362]: Should we implement this loop in CLI?  I can probably consume all of the RPC workers with it here

Comment (sakridge) [reply client/src/rpc_client.rs:362]: client-side workers?

File: cli/src/cli.rs (hunk 3 above)
Comment (sakridge) [reply cli/src/cli.rs]: done

File: system-test/automation_utils.sh (hunk 2 above)
Comment (sakridge) [reply system-test/automation_utils.sh]: yes, done

File: client/src/rpc_client.rs (hunk 4 above)
Comment (t-nelson) [reply client/src/rpc_client.rs:362]: Ah it's just a helper. N/m all good!

**Pull request merged**